src/eth/sync_parser.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashMap;
+
+use ethereum_types::H256;
+use ethers_core::types::TransactionReceipt;
+
+
+pub fn parse_receipt(receipt_json: &str) -> anyhow::Result<HashMap<H256, TransactionReceipt>> {
+    let receipt: TransactionReceipt = serde_json::from_str(receipt_json).map_err(|e| anyhow::anyhow!("Failed to parse receipt JSON: {:?}", e))?;
+
+    let mut receipts_map = HashMap::new();
+    receipts_map.insert(receipt.transaction_hash, receipt);
+
+    Ok(receipts_map)
+}
+
+use ethers_core::types::Block as ECBlock;
+use ethers_core::types::Transaction as ECTransaction;
+
+pub fn parse_block(block_json: &str) -> anyhow::Result<ECBlock<ECTransaction>> {
+    let block: ECBlock<ECTransaction> = serde_json::from_str(block_json)?;
+    Ok(block)
+}
+
+#[cfg(test)]
+mod tests {
+    use ethers_core::k256::U256;
+    use ethers_core::types::H256;
+    use ethers_core::types::U64;
+    use hex_literal::hex;
+
+    use super::*; // Import everything from the outer module
+
+    #[test]
+    fn test_parse_receipts() {
+        // This JSON string represents a single transaction receipt. For testing, it needs to be wrapped in an array.
+        let single_receipt_json = r#"
```

Comment:
Put the JSON in an fixture file and include it with include_str macro to make these tests easier to read.